mapie/time_series_regression.py

```diff
@@ -0,0 +1,325 @@
+                betas_0 = np.full_like(alpha_np, np.nan, dtype=float)
+                for ind, _alpha in enumerate(alpha_np):
+                    betas = np.linspace(
+                        _alpha / (n + 1), _alpha, num=n + 1, endpoint=False
+                    )
+
+                    one_alpha_beta = masked_quantile(
+                        ma.masked_invalid(self.conformity_scores_),
+                        1 - _alpha + betas,
+                        axis=0,
+                        method="higher",
+                    )  # type: ignore
+
+                    beta = masked_quantile(
+                        ma.masked_invalid(self.conformity_scores_),
+                        betas,
+                        axis=0,
+                        method="lower",
+                    )  # type: ignore
+
+                    betas_0[ind] = betas[
+                        np.argmin(one_alpha_beta - beta, axis=1)
+                    ]
+
+                lower_quantiles = masked_quantile(
+                    ma.masked_invalid(self.conformity_scores_),
+                    betas_0,
+                    axis=0,
+                    method="lower",
+                ).T  # type: ignore
+                higher_quantiles = masked_quantile(
+                    ma.masked_invalid(self.conformity_scores_),
+                    1 - alpha_np + betas_0,
+                    axis=0,
+                    method="higher",
+                ).T  # type: ignore
+
+                if (self.method in ["naive", "base"]) or (self.cv == "prefit"):
+                    y_pred_low = np.column_stack(
+                        [
+                            y_pred[:, np.newaxis] + lower_quantiles[k]
+                            for k in range(len(alpha_np))
+                        ]
+                    )
+                    y_pred_up = np.column_stack(
+                        [
+                            y_pred[:, np.newaxis] + higher_quantiles[k]
+                            for k in range(len(alpha_np))
+                        ]
+                    )
+                else:
+                    y_pred_multi = np.column_stack(
+                        [e.predict(X) for e in self.estimators_]
+                    )
+
+                    # At this point, y_pred_multi is of shape
+                    # (n_samples_test, n_estimators_). The method
+                    # ``aggregate_with_mask`` fits it to the right size thanks
+                    # to the shape of k_.
+
+                    y_pred_multi = self.aggregate_with_mask(
+                        y_pred_multi, self.k_
+                    )
+
+                    if self.method == "plus":
+                        pred = aggregate_all(self.agg_function, y_pred_multi)
+                        y_pred_low = np.column_stack(
+                            [
+                                pred + lower_quantiles[k]
+                                for k in range(len(alpha_np))
```

Comment:
Prefer `for k, _ in enumerate(alpha_np)`

Reply:
du coup, plus besoin?

Reply:
Done